File: lib/src/domain/models/budget_model.dart
import 'package:equatable/equatable.dart';

final class BudgetModel extends Equatable {
  final int? id;
  final double amount;
  final int startDate;
  final int endDate;
  final String? description;

  const BudgetModel({
    required this.amount,
    required this.startDate,
    required this.endDate,
    this.id,
    this.description,
  });

  @override
  List<Object?> get props => [id, amount, startDate, endDate, description];
}

File: app/Budget/Infrastructure/Persistence/Models/BudgetModel.php
<?php

declare(strict_types=1);

namespace App\Budget\Infrastructure\Persistence\Models;

use Illuminate\Database\Eloquent\Model;

final class BudgetModel extends Model
{
    protected $table = 'budgets';

    protected $fillable = [
        'amount',
        'end_date',
        'start_date',
        'recurrence_id',
    ];

    protected function casts(): array
    {
        return [
            'amount' => 'integer',
            'end_date' => 'immutable_date',
            'start_date' => 'immutable_date',
            'recurrence_id' => 'integer',
            'created_at' => 'immutable_datetime',
            'updated_at' => 'immutable_datetime',
        ];
    }
}
